compiler/rustc_const_eval/src/interpret/visitor.rs

```diff
@@ -43,7 +40,7 @@ pub trait ValueVisitor<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>>: Sized {
     }
     /// Visits the given value as a union. No automatic recursion can happen here.
     #[inline(always)]
-    fn visit_union(&mut self, _v: &Self::V, _fields: NonZeroUsize) -> InterpResult<'tcx> {
```

Comment:
Just because the currently existing visitors don't need this information, doesn't mean we won't have a visitor that needs this in the future. It'd be annoying to keep removing and adding function arguments as the way this trait is used changes, so think it'd make more sense to keep the parameter around.

Reply:
make sense, rollbacked it.